app/views/students/_form.html.erb

```diff
@@ -11,47 +11,73 @@
     </div>
   <% end %>
 
-  <div class="field">
-    <%= form.label :first_name %>
-    <%= form.text_field :first_name, id: :student_first_name %>
-  </div>
-
-  <div class="field">
-    <%= form.label :last_name %>
-    <%= form.text_field :last_name, id: :student_last_name %>
-  </div>
-
-  <div class="field">
-    <%= form.label :preferred_name %>
-    <%= form.text_field :preferred_name, id: :student_preferred_name %>
-  </div>
-
-  <div class="field">
-    <%= form.label :grad_year %>
-    <%= form.number_field :grad_year, id: :student_grad_year %>
-  </div>
-
-  <div class="field">
-    <%= form.label :osis %>
-    <%= form.number_field :osis, id: :student_osis %>
-  </div>
-
-  <div class="field">
-    <%= form.label :email %>
-    <%= form.text_field :email, id: :student_email %>
-  </div>
-
-  <div class="field">
-    <%= form.label :phone %>
-    <%= form.text_field :phone, id: :student_phone %>
-  </div>
-
-  <div class="field">
-    <%= form.label :team_id %>
-    <%= form.text_field :team_id, id: :student_team_id %>
-  </div>
-
-  <div class="actions">
-    <%= form.submit %>
-  </div>
+  <% if current_user&.is_admin %>
```

Comment:
What might make this cleaner is separate partials for admin-editable fields and user-editable fields.

You can make these partials under `app/views/students/form/` and render them from a conditional in `_form.html.erb`:

```
<% if current_user&.is_admin %>
  render the admin form here
<% elsif (@user == @record&.user) %>
  render the limited form here
<% end %>
```